WordPress/Classes/ViewRelated/Me/My Profile/Change Username/ChangeUsernameViewController.swift

```diff
@@ -31,6 +32,13 @@ class ChangeUsernameViewController: SignupUsernameTableViewController {
         super.viewDidLoad()
         setupViewModel()
         setupUI()
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(handleTextDidChangeNotification(_:)),
+            name: UITextField.textDidChangeNotification,
+            object: nil
+        )
```

Comment:
This feels _somewhat_ fragile to me in the context of `UIScene`, because (IINM) we're capturing the `textDidChangeNotification` for every `UITextField` in the currently running app. Is that correct?

If so, this is a good bug fix, but adopts a different kind of technical debt, so I wonder if we ought to approach it slightly differently? 🤔

Reply:
Good point. There is no need to observe all text change during the lifecyle of this view controller, even though the notification handler does nothing for notifications that it doesn't care about.

I have changed the approach to only observe the text field in the alert: https://github.com/wordpress-mobile/WordPress-iOS/pull/21003/commits/69b707af3e89f66a0e2da03ca5cac3eb9a5f5346.